vms/platformvm/service.go

```diff
@@ -2143,48 +2405,57 @@ func (service *Service) GetStake(_ *http.Request, args *GetStakeArgs, response *
 		addrs.Add(addr)
 	}
 
-	currentStakers := service.vm.internalState.CurrentStakerChainState()
-	stakers := currentStakers.Stakers()
-
+	stakers := service.vm.internalState.CurrentStakerChainState().Stakers()
 	var (
 		totalStake uint64
 		stakedOuts = make([]avax.TransferableOutput, 0, len(stakers))
+		stakeParts = [3]uint64{0, 0, 0}
 	)
-	for _, tx := range stakers { // Iterates over current stakers
-		stakedAmt, outs, err := service.getStakeHelper(tx, addrs)
-		if err != nil {
-			return err
-		}
-		totalStake, err = math.Add64(totalStake, stakedAmt)
-		if err != nil {
-			return err
-		}
-		stakedOuts = append(stakedOuts, outs...)
-	}
-
-	pendingStakers := service.vm.internalState.PendingStakerChainState()
-	for _, tx := range pendingStakers.Stakers() { // Iterates over pending stakers
-		stakedAmt, outs, err := service.getStakeHelper(tx, addrs)
-		if err != nil {
-			return err
+	for i := 0; i < 3; i++ {
```

Comment:
This whole thing with for-loop and if-s inside looks messy and hard to understand. I'd rewrite it